index.js

```diff
@@ -125,16 +125,17 @@ async function fastifyRateLimit (fastify, settings) {
 
   fastify.decorateRequest(pluginComponent.rateLimitRan, false)
 
+  if (!fastify.hasDecorator('createRateLimit')) {
+    fastify.decorate('createRateLimit', (options) => {
+      const args = createLimiterArgs(pluginComponent, globalParams, options)
+      return (req) => applyRateLimit(...args, req)
```

Comment:
Instead of decostruct `args` we can:
```suggestion
      return (req) => applyRateLimit.apply(this, args.concat(req))
```

(and even better, avoid the `concat`